sql/expression/function/greatest_least.go

```diff
@@ -0,0 +1,310 @@
+package function
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// compEval is used to implement Greatest/Least Eval() using a comparison function
+func compEval(
+	returnType sql.Type,
+	args []sql.Expression,
+	ctx *sql.Context,
+	row sql.Row,
+	cmp compareFn,
+) (interface{}, error) {
+
+	if returnType == sql.Null {
+		return nil, nil
+	}
+
+	var selectedNum float64
+	var selectedString string
+
+	for i, arg := range args {
+		val, err := arg.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+
+		switch t := val.(type) {
+		case int, int32, int64:
```

Comment:
What about `unsigned` variants?
I don't want to be too much `tiquismiquis`, but maybe we should stick just to 2 options - `string` or `float64`.
I remember we had the same _pleasure_ fighting with different types for arithmetic functions (max, min, avg, ...) and ultimately we decided to stick to float64.

Reply:
Well, there is not much difference between supporting only `string` and `float64` and also supporting `int/32/64` but not unsigned.

Reply:
I just meant:
```
if string {
// convert to string
} else {
// convert to float64
}
```

if we wanna keep the current implementation it's fine, if we add int8, int16 and unsigned variants

Reply:
@kuba [PTAL](https://github.com/src-d/go-mysql-server/pull/700/commits/7cc58ce9b8e84cd84299e1946b5d77f0934a1899)